docs/cli/ilab-model-management-standardization.md

```diff
@@ -0,0 +1,81 @@
+- **Scope**: Covers model directory structure in cache and all model-related flags used in the command-line interface (CLI) and libraries within the project.
+- **Audience**: Developers, maintainers, and contributors involved in the development and usage of the CLI tools and libraries.
+
+## Summary
+
+This proposal aims to establish a consistent practice for how models are managed within InstructLab. It does so in 2 parts - one proposing a consolidation of all existing model related flags, and the other proposing a change to how models are organized in the cache upon download, and thereby referenced within various operations
+
+## Motivation
+
+Currently there are 10+ instances across all major commands and subcommands that accept either `--model` , `--model-path`, `--model-name`, or `--model-dir`; all of which serve slightly different purposes, and might handle different use cases including local vs remote models, relative vs absolute paths etc. This leads to a significant amount of avoidable confusion among developers and users alike. Introducing some consolidation and consistency in this aspect would provide a big UX improvement.
+
+In addition to this, we currently support multiple sources for model downloads. There could be models that span multiple sources and also have different versions and tags that users may want to be able to work with. Currently, there is a lack of uniformity in how we organize models downloaded from Hugging Face vs OCI registries. We do not have a way to differentiate and maintain unique identities for models in cache, based on their sources. We also currently don't have the ability to support version management. Users are only able to work with one version of a given model at a time, as downloading a different version just overwrites their existing model.
+
+## Proposal
+
+### Part 1: Narrowing down to a singular model related flag
+
+#### Problem Statement - Section 1
+
+We want to streamline the number of model related flags available, while establishing consistency and predictability between their uses and handling 3 separate use-cases: absolute paths, relative paths and remote repository names.
+
+#### Suggested approach - Section 1
+
+- Retain `--model` and deprecate all other model related flags
+  - Alternatively, we could retain `--model-path` but passing in a repo name to a flag called "model-path" seems a little more awkward than passing in a path to a field just called "model"
+- `--model` should accept both paths, as well as strings (for repo names)
+- First `--model` should assume whatever is passed into it is an absolute path, and should check against root to determine if that path exists
+  - If it exists, we should run an additional check to determine if it points to a safetensor or gguf models (use existing `is_model_safetensors` and `is_model_gguf` checks for this)
+  - If path does not exist, or exists but is not determined to be a valid model, then move on
```

Comment:
It doesn't matter if this is a relative or absolute path. You can condense the above 2 points to just check if the given path exists. Both cases will be handled with the exact same logic.

Reply:
but by default, wouldn't relative path be checked relative to where the program is running from, rather than relative to `~/.local/cache/instructlab/models`?

Reply:
@jaideepr97 the correct way to handle this would be:

1. Check if the file at the given path exists, return if true
2. Get the contents of model list
3. If the given path matches any of the listed, return here
4. Else error it's not found

Possible input cases:
A. Given path is relative to calling program
B. Given path is an absolute path
C. Given path is a result from the listed
D. Given path doesn't exist 


In case A it will return on step 1
In case B it will also return on step 1
In case C it will return on step 3
In case D it will hit step 4

Reply:
workflow is updated to be:
1. Check if the supplied path is either absolute, or relative to where the program is running
2. check if supplied path is relative to default cache dir
3. if neither - error out or download model from somewhere (TBD)